### ISSUE TITLE
inline comments not supported in configuration file

### ISSUE THREAD

**Issue by thomas-paul-ohara** [PyCQA/isort#2292]
The isort configuration file doesn't support inline comments. For example,

```
$ head .isort.cfg hello.py
==> .isort.cfg <==
[settings]
force_single_line=1                  # every import will appear on its own

==> hello.py <==
import sys
sys.stdout.write("hello\n")

$ isort .
/Users/eafqe/virtual-environments/py-3-11/lib/python3.11/site-packages/isort/settings.py:783: UserWarning: Failed to pull configuration information from /Users/eafqe/temp/Gen-AI-Product-Geo-Extractor-tom/src/isort/bug/.isort.cfg
  warn(f"Failed to pull configuration information from {potential_config_file}")
Fixing /Users/eafqe/temp/Gen-AI-Product-Geo-Extractor-tom/src/isort/bug/hello.py

$ perl -i.bak -pe 's/#/\n#/;' .isort.cfg
$ isort .

$ cat .isort.cfg
[settings]
force_single_line=1
# every import will appear on its own
```

It would be good if comments can be stripped, because inline comments make it simpler to explain settings rather than alternating comments and settings.

**Comment by kurtmckee** [comment]
Inline comments are supported for the `pyproject.toml` file format.

I recommend closing this issue to reduce the backlog of issues to address. Thanks!

**Comment by kurtmckee** [comment]
@DanielNoord This feature request would directly conflict with an existing isort feature, [treat-comments-as-code](https://pycqa.github.io/isort/docs/configuration/options.html#example-isortcfg_19). I recommend closing this issue.

**Comment by DanielNoord** [comment]
Makes sense, but I won't close to see if this gets some support from other users.

**Comment by kurtmckee** [comment]
INI files don't use quotation marks to distinguish between the config value and a trailing comment.

The example for `treat-comments-as-code`, linked above, is:

```
[settings]
treat_comments_as_code = # my comment 1, # my other comment
```

There is no way to parse this -- and trailing comments as requested here -- deterministically from an INI file.

**Comment by thomas-paul-ohara** [comment]
For future work, it would be better to flag this as "won't fix" or "not planned".  I consider it a regular bug rather than an enhancement request, because a warning isn't issued. It took a lot of time to track down the source of why "force_single_line=1" was being ignored.

Also note that having developers close tickets themselves as "not planned" adds no burden regarding the backlog of issues to be addressed. Shaming users into doing so shows a lack of appreciation of the effort put into the bug report.

**Comment by kurtmckee** [comment]
@thomas-paul-ohara My apologies, I wasn't attempting to shame you into closing this issue. The maintainers weren't active for a very long time, so I set a goal to inject energy to the project via rapid-fire triage. I myself have no ability to close tickets myself, hence the request.

I apologize again and hope you'll continue contributing to isort in the future.
